### PR TITLE
feat: add single selection filter input type and operator

### PR DESCRIPTION
**Pull request by jogeshwar01**
Closes: #8138 

### Description:
Added an `EQUALS_ONLY` FilterOperator

There are basically 2 main types for which we would need this – `STRING` and `NUMBER`.
For these, I have kept the backend same ie. using the IN clause in sql, just that now we would have a single element array when EQUALS_ONLY is selected.

For the frontend,
the numbers can directly use our `FilterNumberInput` which accepts a simple `NumberInput`.

In strings, for cases (like EQUALS, NOT_EQUALS etc), we use the components `FilterStringAutoComplete` and `FilterMultiStringInput` depending on whether it `isTableCalculationField`.
Modifying both of these components was also an alternative here (but that might have made the logic more complex)

So, I have added a new component `FilterSingleStringAutoComplete` as this felt a lot cleaner to me.
This handles both cases - Normal as well as TableCalculationFields using a flag. 
Here we would have a Select rather than a Multi-Select

I have also added some logic in useEffect to handles option changes from multiple to single inputs.
This is useful when a user has selected lets say 3 inputs (in `is` option) and then switches to `is_only` option. Then we update the value with the first element of the input. I do think there will be an optimised way to do this. Would love to discuss this.

Here are some screenshots of the output,

Strings -
image

image

Numbers -
image

image

Combination - 
image

image

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx [hunk 1]
```diff
@@ -119,20 +121,55 @@ const DefaultFilterInputs = <T extends ConditionalRule>({
...
-                        onChange({
-                            ...rule,
-                            values: newValue ? [newValue] : [],
-                        });
-                    }}
-                />
-            );
+        case FilterOperator.IN_BETWEEN: {
+            switch (filterType) {
+                case FilterType.STRING:
```
Comment (joaoviana) [inline packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx:126]: perhaps this (and the case in 115) can be moved to its own `case`? I think it's a bit confusing 🤔

Comment (joaoviana) [reply packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx:139]: Thank you @jogeshwar01 for the prompt push. I will review this locally and with a duplicate PR to test e2es :)

**Review by joaoviana — CHANGES_REQUESTED** [review]
Looks great @jogeshwar01 !!

I've tested so far: 
* Charts + filters
* Dashboard + saved filters + temporary filters + overrides + exports + scheduled deliveries (send now and scheduled)

Could you add this change?

In: packages/frontend/src/components/DashboardFilter/FilterConfiguration/utils/index.ts

image

**Comment by joaoviana** [comment]
One for maybe @TuringLovesDeathMetal 
I add filter with value `x` with the `is only` operator
I try to add temporary filter with value `y` with the `is only` operator as well
It doesn't let me choose the other values. 

image

Expected? I think so, makes sense

**Comment by jogeshwar01** [comment]
> Could you add this change?
> 
> In: packages/frontend/src/components/DashboardFilter/FilterConfiguration/utils/index.ts

@joaoviana 
Ah, I shouldn't have missed that. Thanks for pointing it out.
I have added this change.

**Comment by TuringLovesDeathMetal** [comment]
Yep, that behaviour makes sense to me!

**Comment by joaoviana** [comment]
Hi @jogeshwar01 , thanks so much for the contribution on this issue. 
We've discussed this internally, and although an operator like `is only` would allow for single selection filtering, we believe that the current `equals/is` filter type input should be extended/reworked to allow only single selection. This requires a bit of thinking with Product, Design, and other customers/users. 
Unfortunately I have to close this PR.

**Comment by jogeshwar01** [comment]
@joaoviana Appreciate the review. I'm on board with adjusting the current equals/is filter input for single selection. I'll be happy to contribute on this once the proposal is finalized.

**Pull request closed without merging**